### PR TITLE
feat: customize name of Shield Tables w/o constant

### PR DESCRIPTION
**Pull request by kenjis**
Ref https://github.com/codeigniter4/shield/pull/628#issuecomment-1424971414
Supersedes #628

### CODE REVIEW

**Comment by datamweb** [comment]
Please update the branch, we have recently made changes:
https://github.com/codeigniter4/shield/pull/629/files#diff-72d57325d612cc25a56eec53a2f8c721d8ef92fd14b20e2f83446f226972f715R476

File: src/Models/UserModel.php [hunk 1]
```diff
@@ -49,6 +49,20 @@ class UserModel extends Model
...
+     */
+    private string $tableIdentities;
+
+    protected function initialize(): void
+    {
+        /** @var Auth $authConfig */
+        $authConfig = config('Auth');
+
+        $this->table           = $authConfig->tables['users'];
+        $this->tableIdentities = $authConfig->tables['identities'];
```
Comment (datamweb) [inline src/Models/UserModel.php:63]: I didn't like this, I prefer it as below to maintain the consistency of using table names throughout the code.
```php
$this->table           = $authConfig->tables['users'];
$this->tables['users'] = $authConfig->tables['users'];
$this->tables['identities'] = $authConfig->tables['identities'];
```

Comment (kenjis) [reply src/Models/UserModel.php]: Done.

File: docs/customization.md [hunk 2]
```diff
@@ -150,11 +150,11 @@ Shield has the following rules for registration:
 [
     'username' => [
         'label' =>  'Auth.username',
-        'rules' => 'required|max_length[30]|min_length[3]|regex_match[/\A[a-zA-Z0-9\.]+\z/]|is_unique[users.username]',
+        'rules' => 'required|max_length[30]|min_length[3]|regex_match[/\A[a-zA-Z0-9\.]+\z/]|is_unique[' . SHIELD_TABLES['users'] . '.username]',
```
Comment (datamweb) [inline docs/customization.md:153]: Please fix it.

File: docs/customization.md [hunk 3]
```diff
@@ -150,11 +150,11 @@ Shield has the following rules for registration:
...
     ],
     'email' => [
         'label' =>  'Auth.email',
-        'rules' => 'required|max_length[254]|valid_email|is_unique[auth_identities.secret]',
+        'rules' => 'required|max_length[254]|valid_email|is_unique[' . SHIELD_TABLES['identities'] . '.secret]',
```
Comment (datamweb) [inline docs/customization.md:157]: and

File: docs/customization.md [hunk 4]
```diff
@@ -176,11 +176,11 @@ If you need a different set of rules for registration, you can specify them in y
     public $registration = [
         'username' => [
             'label' =>  'Auth.username',
-            'rules' => 'required|max_length[30]|min_length[3]|regex_match[/\A[a-zA-Z0-9\.]+\z/]|is_unique[users.username]',
+            'rules' => 'required|max_length[30]|min_length[3]|regex_match[/\A[a-zA-Z0-9\.]+\z/]|is_unique[' . SHIELD_TABLES['users'] . '.username]',
```
Comment (datamweb) [inline docs/customization.md:179]: and

File: docs/customization.md [hunk 5]
```diff
@@ -176,11 +176,11 @@ If you need a different set of rules for registration, you can specify them in y
...
         ],
         'email' => [
             'label' =>  'Auth.email',
-            'rules' => 'required|max_length[254]|valid_email|is_unique[auth_identities.secret]',
+            'rules' => 'required|max_length[254]|valid_email|is_unique[' . SHIELD_TABLES['identities'] . '.secret]',
```
Comment (datamweb) [inline docs/customization.md:183]: and

File: rector.php [hunk 6]
```diff
@@ -99,6 +99,7 @@
             __DIR__ . '/src/Helpers',
             __DIR__ . '/src/Language',
             __DIR__ . '/tests/_support',
+            __DIR__ . '/src/Config/Constants.php',
```
Comment (datamweb) [inline rector.php:102]: deleted this.

File: src/Models/GroupModel.php [hunk 7]
```diff
@@ -4,14 +4,10 @@
 
 namespace CodeIgniter\Shield\Models;
 
-use CodeIgniter\Model;
 use CodeIgniter\Shield\Entities\User;
 
-class GroupModel extends Model
+class GroupModel extends BaseModel
```
Comment (datamweb) [inline src/Models/GroupModel.php:9]: Isn't it better to declare the use of BaseModel?
```php
use CodeIgniter\Shield\Models\BaseModel;
```

File: src/Models/LoginModel.php [hunk 8]
```diff
@@ -5,17 +5,13 @@
...
 
 use CodeIgniter\I18n\Time;
-use CodeIgniter\Model;
 use CodeIgniter\Shield\Authentication\Authenticators\Session;
 use CodeIgniter\Shield\Entities\Login;
 use CodeIgniter\Shield\Entities\User;
 use Faker\Generator;
 
-class LoginModel extends Model
+class LoginModel extends BaseModel
```
Comment (datamweb) [inline src/Models/LoginModel.php:13]: The same as the previous comment.

File: src/Models/PermissionModel.php [hunk 9]
```diff
@@ -4,14 +4,10 @@
 
 namespace CodeIgniter\Shield\Models;
 
-use CodeIgniter\Model;
 use CodeIgniter\Shield\Entities\User;
 
-class PermissionModel extends Model
+class PermissionModel extends BaseModel
```
Comment (datamweb) [inline src/Models/PermissionModel.php:9]: The same as the previous comment

File: src/Models/RememberModel.php [hunk 10]
```diff
@@ -5,17 +5,13 @@
...
 
 use CodeIgniter\I18n\Time;
-use CodeIgniter\Model;
 use CodeIgniter\Shield\Entities\User;
 use DateTime;
 use Faker\Generator;
 use stdClass;
 
-class RememberModel extends Model
+class RememberModel extends BaseModel
```
Comment (datamweb) [inline src/Models/RememberModel.php:13]: The same as the previous comment.

File: src/Models/TokenLoginModel.php [hunk 11]
```diff
@@ -10,7 +10,12 @@
 
 class TokenLoginModel extends LoginModel
```
Comment (datamweb) [inline src/Models/TokenLoginModel.php:11]: The same as the previous comment.
```php
use CodeIgniter\Shield\Models\LoginModel;
```

File: src/Models/UserIdentityModel.php [hunk 12]
```diff
@@ -16,11 +15,8 @@
 use CodeIgniter\Shield\Exceptions\ValidationException;
 use Faker\Generator;
 
-class UserIdentityModel extends Model
+class UserIdentityModel extends BaseModel
```
Comment (datamweb) [inline src/Models/UserIdentityModel.php:18]: and

File: src/Models/UserModel.php (hunk 1 above)
Comment (datamweb) [reply src/Models/UserModel.php]: Thanks!

File: src/Models/GroupModel.php (hunk 7 above)
Comment (kenjis) [reply src/Models/GroupModel.php:9]: Why?
`BaseModel` is in the same namespace of the `GroupModel`.
So the `use` statement is not needed.

File: docs/customization.md (hunk 2 above)
Comment (kenjis) [reply docs/customization.md]: Done.

File: docs/customization.md (hunk 3 above)
Comment (kenjis) [reply docs/customization.md]: Done.

File: docs/customization.md (hunk 4 above)
Comment (kenjis) [reply docs/customization.md]: Done.

File: docs/customization.md (hunk 5 above)
Comment (kenjis) [reply docs/customization.md]: Done.

File: rector.php (hunk 6 above)
Comment (kenjis) [reply rector.php]: Done.

**Comment by kenjis** [comment]
Added docs.

**Review by datamweb — APPROVED** [review]
LGTM!
thanks!

**Pull request merged**